### PR TITLE
New Features for Developing AngularJS Applications - Part 1: Component-Based Architecture

### PR DESCRIPTION
**Pull request by unicodeveloper**

### CODE REVIEW

File: _posts/2018-01-18-developing-angularjs-applications-part-1.markdown [hunk 1]
```diff
@@ -0,0 +1,579 @@
...
+{
+  ...
+  scope: {},
+  ...
+}
+```
+
+## Stateless components
+
+There’s now the ability to create “stateless” components, read my in-depth article on stateless components in the .component() method.
```
Comment (sebadoom) [inline _posts/2018-01-18-developing-angularjs-applications-part-1.markdown]: What article?

File: _posts/2018-01-18-developing-angularjs-applications-part-1.markdown [hunk 2]
```diff
@@ -0,0 +1,579 @@
...
+    </div>
+  `
+};
+
+angular
+  .module('app', [])
+  .component('nameComponent', NameComponent);
+
+```
+
```
Comment (sebadoom) [inline _posts/2018-01-18-developing-angularjs-applications-part-1.markdown]: Aside?

File: _posts/2018-01-18-developing-angularjs-applications-part-1.markdown [hunk 3]
```diff
@@ -0,0 +1,579 @@
...
+  scope: {},
+  ...
+}
+```
+
+## Stateless components
+
+There’s now the ability to create “stateless” components, read my in-depth article on stateless components in the .component() method.
+
+Essentially we can just use a template and bindings:
```
Comment (sebadoom) [inline _posts/2018-01-18-developing-angularjs-applications-part-1.markdown]: This part needs a better description.

File: _posts/2018-01-18-developing-angularjs-applications-part-1.markdown [hunk 4]
```diff
@@ -0,0 +1,579 @@
...
+      </div>
+    `
+  }
+  ...
+}
+```
+
+## One Way Data Binding
+
+If you have been using AngularJS long enough, you will be accustomed to the two-data way binding it provides. However, in AngularJS 1.5+, there is a new one-way data binding that we can explore and also improve the performance of our AngularJS applications.
```
Comment (sebadoom) [inline _posts/2018-01-18-developing-angularjs-applications-part-1.markdown]: Add a quick refresher on what one-way/two-way binding is.

File: _posts/2018-01-18-developing-angularjs-applications-part-1.markdown [hunk 5]
```diff
@@ -0,0 +1,579 @@
...
+```js
+// 1.5
+{
+  ...
+  controller: 'SomeCtrl as something'
+  ...
+}
+```
+
+This saves adding the controllerAs property… however…
```
Comment (sebadoom) [inline _posts/2018-01-18-developing-angularjs-applications-part-1.markdown]: Is there something missing here?

File: _posts/2018-01-18-developing-angularjs-applications-part-1.markdown [hunk 6]
```diff
@@ -0,0 +1,579 @@
...
+- 2017-10-12-consolidating-multiple-identity-sources-with-auth0
+- 2017-10-13-how-do-i-modernize-my-legacy-system
+---
+
+**TL;DR:** AngularJS 1.5+ has undergone major updates. In this article, you'll learn how to develop AngularJS applications using a component-based architecture and leverage the API changes that will help you write your AngularJS applications in an Angular(v2+) style.
+
+The Component-based Architecture is prevalent in frontend development today. JavaScript Client-side frameworks such as React, Preact, Ember, Vue.js all advocate building applications in components. _Angular_ also popularly known as _Angular(v2+)_ which is a major rewrite of AngularJS is completely component-based.
+
+A typical example of an application built applying the component architecture involves developing several features of the application as modules with reusable components. For example, let's use a Music Player application. A required feature for this product could be a Music Player widget. A developer designing their app in a component-based architecture style can model this feature as a *Module*.
+
```
Comment (sebadoom) [inline _posts/2018-01-18-developing-angularjs-applications-part-1.markdown]: Add a short notice here that this article assumes the reader is already familiar with AngularJS and is not an introductory article (like the title suggests).

File: _posts/2018-01-18-developing-angularjs-applications-part-1.markdown [hunk 7]
```diff
@@ -0,0 +1,579 @@
+---
+layout: post
+title: "Developing AngularJS Applications - Part 1: Component-Based Architecture"
```
Comment (sebadoom) [inline _posts/2018-01-18-developing-angularjs-applications-part-1.markdown]: Hmm, I think "Updating AngularJS Applications" or "New Features for Developing AngularJS Applications" is more apt than "Developing". Ask in #mktg_content about this, this article reads like a tutorial on how to update your old Angular apps to newer ideas from Angular 1.5+ and not a tutorial on how to develop Angular apps from scratch.

File: _posts/2018-01-18-developing-angularjs-applications-part-1.markdown [hunk 8]
```diff
@@ -0,0 +1,579 @@
...
+
+_component_
+
+```js
+.component('counter', function counter() {
+
+});
+```
+
+Nice and simple. Essentially the return {}; statement inside the .directive() becomes the Object definition inside .component() - easy!
```
Comment (sebadoom) [inline _posts/2018-01-18-developing-angularjs-applications-part-1.markdown]: Use `inline code tags` for `.component()` and other stuff around this article, they are missing in many places.

File: _posts/2018-01-18-developing-angularjs-applications-part-1.markdown [hunk 9]
```diff
@@ -0,0 +1,579 @@
...
+  };
+}
+
+angular
+  .module('app', [])
+  .component('sample', sample)
+  .controller('ExampleController', ExampleController);
+```
+
+```bash
```
Comment (brunokrebs) [inline _posts/2018-01-18-developing-angularjs-applications-part-1.markdown]: bash?

Comment (brunokrebs) [reply _posts/2018-01-18-developing-angularjs-applications-part-1.markdown]: please, use this: https://github.com/auth0/blog/blob/master/post-cheat-sheet.markdown#code-syntax-highlighting

File: _posts/2018-01-18-developing-angularjs-applications-part-1.markdown [hunk 10]
```diff
@@ -0,0 +1,579 @@
...
+        }
+      },
+      ...
+
+    ];
+  };
+}
+```
+
+```bash
```
Comment (brunokrebs) [inline _posts/2018-01-18-developing-angularjs-applications-part-1.markdown]: same here: https://github.com/auth0/blog/blob/master/post-cheat-sheet.markdown#code-syntax-highlighting

**Pull request merged**